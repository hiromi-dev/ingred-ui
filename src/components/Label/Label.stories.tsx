import * as React from "react";
import { Story } from "@storybook/react/types-6-0";
import Label, { LabelProps } from ".";

export default {
  title: "Components/Data Display/Label",
  component: Label,
  args: {
    labelTitle: "sample",
  },
};

export const Example: Story<LabelProps> = (args) => <Label {...args} />;

export const CloseButton: Story<LabelProps> = (args) => {
  const handleRemove = (labelEleent: any) => {};
  return (
    <Label
      labelElement={{ name: "sample" }}
      onRemove={handleRemove}
      {...args}
    />
  );
};
