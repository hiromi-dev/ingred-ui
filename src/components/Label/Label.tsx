import * as React from "react";
import Typography from "../Typography";
import { FontSize } from "../Typography/Typography";
import { CloseButton } from "./internal/CloseButton";
import * as Styled from "./styled";

export type LabelProps = {
  labelTitle?: string;
  labelElement?: any;
  onRemove?: (labelElement: any) => void;
  size?: FontSize;
};

const Label: React.FunctionComponent<LabelProps> = ({
  labelTitle,
  labelElement,
  onRemove,
  size = "sm",
}) => {
  const handleClick = () => {
    if (onRemove) {
      onRemove(labelElement);
    }
  };
  return (
    <Styled.Container>
      <Styled.LeftContainer>
        <Typography size={size} component="span">
          {labelTitle}
        </Typography>
      </Styled.LeftContainer>
      {onRemove ? (
        <Styled.RightContainer>
          <CloseButton onClick={handleClick} />
        </Styled.RightContainer>
      ) : (
        <div />
      )}
    </Styled.Container>
  );
};

export default Label;
