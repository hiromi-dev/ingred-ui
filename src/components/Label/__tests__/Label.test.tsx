import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import Label from "../";

describe("Label component testing", () => {
  afterEach(cleanup);

  test("Label", () => {
    const { asFragment } = renderWithThemeProvider(<Label />);
    expect(asFragment()).toMatchSnapshot();
  });
});
