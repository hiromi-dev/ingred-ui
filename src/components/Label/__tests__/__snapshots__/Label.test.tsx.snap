// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component testing Label 1`] = `
<DocumentFragment>
  <div
    class="sc-hKgILt jCfqnH"
  >
    <div
      class="sc-eCssSg lbcvem"
    >
      <span
        class="sc-bdfBwQ gNlqto"
        color="#041C33"
        font-size="13px"
      />
    </div>
    <div />
  </div>
</DocumentFragment>
`;
